pkg/pillar/cmd/tpmmgr/handlecertificate.go

```diff
@@ -0,0 +1,301 @@
+// Copyright (c) 2018-2019 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package tpmmgr
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"os"
+
+	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpmutil"
+	"github.com/lf-edge/eve/api/go/evecommon"
+	etpm "github.com/lf-edge/eve/pkg/pillar/evetpm"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	log "github.com/sirupsen/logrus"
+)
+
+// update the device generated certificates on pubsub
+func publishEveNodeCertificates(ctx *tpmMgrContext) error {
+
+	// ECDH Certificate
+	certType := evecommon.ZCertType_Z_CERT_TYPE_DEVICE_ECDH_EXCHANGE
+	if err := publishEveNodeCertificate(ctx, certType); err != nil {
+		return err
+	}
+
+	// TBD:XXX for attest and others
+	return nil
+}
+
+// generate/publish certificate, of a given type
+func publishEveNodeCertificate(ctx *tpmMgrContext,
```

Comment:
Function name is misleading. I think separate create and publish functions makes it more clear.
And I expect them to handle more than one certificate soon, hence there is no need to pass a type, but it makes more sense to have a 
  generateEveCertificates() and a separate publishEveCertificates() which will also do the attestation cert when that is added. Hence no need to pass a certType here; that is harmful to expanding this code.